.github/workflows/lint.yml

```
@@ -0,0 +1,60 @@
+name: Lint Rust files
+
+on:
+  push:
+    branches:
+      - "**"
+      - "!main"
+    path:
+      - '**/*.rs'
+      - '**/Cargo.toml'
+      - '**/Cargo.lock'
+      - 'clippy.toml'
+      - '.cargo/config.toml'
+      - '.github/workflows/lint.yml'
+
+env:
+  CARGO_INCREMENTAL: 0
+  RUST_BACKTRACE: full
+
+jobs:
+  clippy:
+    name: Clippy
```

This job name is one of the required jobs in our repository GitHub config, so it's easier if we keep it the same.

```suggestion
    name: Clippy (stable)
```

---

Actually, one of the things I'll be changing is removing the special characters from the job's name, including parenthesis, plus signs, etc. I had issues when building a regex using Mergify, and it might happen with other tools.

I'll do the required changes in the repo configuration to make this new job the required one, and removed the old one (as it won't get reported).

For future changes which require dynamic names (stable, nightly, etc) using the job matrix and adding it to the name is a better option. But as of now, this is not the case with Clippy. It also makes you wonder if there's another Clippy with nightly checks.

---

Sounds good, thanks for explaining, the pending status on the PR was a bit confusing.

> It also makes you wonder if there's another Clippy with nightly checks.

There was, but we never re-enabled it. And I don't think we're going to do that any time soon.